R/checks.R

```diff
@@ -0,0 +1,425 @@
+#' Complete initial checks for submissions.
+#'
+#' There are several initial checks undertaken for each submission.
+#' These checks reflect the requests made in the author submission guide.
+#'
+#' Results will be show in the initial_checks.log file in the submission folder
+#'
+#' Set the directory of the articles repository using `set_articles_path`.
+#'
+#'
+initial_checks <- function() {
+
+  # Find articles without initial_check logs, pass that list to do checks
+  base <- file.path(get_articles_path(), "Submissions")
+  paths <- dir(base, full.names = TRUE)
+
+  # Call the function for to check each submission
+  lapply(paths, initial_check_article)
+
+  return(cat("Checks completed"))
+}
+
+
+#' A single article check
+#'
+#' There are several initial checks undertaken for a submission.
+#' These checks reflect the requests made in the author submission guide.
+#'
+#' Results will be show in the initial_checks.log file in the submission folder
+#'
+#' Set the directory of the articles repository using `set_articles_path`.
+#'
+#' @param path Location of the article submission folder to check
+#'
+#' @export
+initial_check_article <- function(path) {
+
+  # Documents:
+  # Necessary files must be included in submission folder
+  # Do not proceed without them, flag for manual check
+  # Unnecessary files must not be included in submission folder
+  # Do not proceed with them, flag for manual check
+
+  # Create a log file for errors
+  sink(file.path(path, "initial_checks.log"))
+
+  # Display the name of the current article
+  cat(paste0("Initial check results: ", basename(path), "\n"))
+
+  cli::cli_h1(paste0("Initial check results: ", basename(path)))
+
+  # Find all file names in the submission folder
+  # for single submission in submissions folder
+  submission_files <- list.files(path)
+
+
+  remaining_files <- check_wrappers(submission_files)
+
+
+  # BEGIN CHECKS
+  # Check for one of each  "tex", "bib", "R"
+  # return the name of the tex file
+  filename <- check_filenames(remaining_files)
+
+
+  # Check for unnecessary files (to avoid potential issues)
+  check_unnecessary_files(submission_files)
+
+  # Check for existence of a cover letter
+  check_cover_letter(remaining_files)
+
+  # check that packages mentioned in manuscript are available on CRAN
+  # if there are no mentions, flag for manual check
+  available(path, filename)
```

Comment:
Prefix check functions with `check_*()` can help identify the purpose of the function. `available()` is too generic (availability of the paper? package? etc.)

Reply:
Great change, thanks!